gtfs-realtime/spec/en/trip-updates.md

```diff
@@ -25,6 +25,10 @@ For each [StopTimeUpdate](reference.md#StopTimeUpdate), the default schedule rel
 
 If one or more stops are missing along the trip the update is propagated to all subsequent stops. This means that updating a stop time for a certain stop will change all subsequent stops in the absence of any other information.
 
+Delay values (including both a [`StopTimeEvent.delay`](reference.md#message-stoptimeevent) as well as delays calculated by applying a [`StopTimeEvent.time`](reference.md#message-stoptimeevent)) to the schedule in [GTFS `stop_times.txt`](/gtfs/spec/en/reference.md#stop_timestxt) greater than or equal to `0` (vehicles running late or on time) must be propagated across trips within the same block (defined by [GTFS trips.txt block_id](/gtfs/spec/en/reference.md#tripstxt)).  If there is a layover in between Trips A and B (e.g., a 5 minute difference between the Trip A final stop `arrival_time` and the Trip B first stop `departure_time`) and no real-time `delay` or `time` information is provided for the Trip B stops, consumers should subtract the layover time from any delay values from Trip A before propagating the delay to Trip B.  This means that consumers will, in the absence of real-time information for Trip B, assume that if the vehicle is running late it will skip it's scheduled layover time and depart on Trip B as soon as it reaches the stop to catch up to the schedule.  If the vehicle is running late on Trip A but plans to hold for the entire scheduled layover time before starting Trip B, producers must represent this by providing [StopTimeUpdates](reference.md#StopTimeUpdate) on Trip B while the vehicle is serving Trip A so riders waiting for Trip B can see that there will be an extended delay. 
+
+Delay values less than `0` (i.e., vehicles running early) must be propagated across trips in the same block if there is no layover between Trip A and Trip B. If there is difference between the Trip A final stop `arrival_time` and the Trip B first stop `departure_time`, then consumers must assume that the vehicle will hold after finishing Trip A and will start Trip B on time. Producers must indicate if a vehicles is departing early on Trip B by explicitly providing a [StopTimeEvent](reference.md#StopTimeEvent) for Trip B while Trip A is being served by the vehicle.
```

Comment:
@LeoFrachet Do you have thoughts on propagating early arrivals across trips in the same block?

I've written this rule to follow OBA behavior, which is:

>    /**
     * If the vehicle is running early and there is slack built into the
     * schedule, we guess that the vehicle will take that opportunity to pause
     * and let the schedule catch back up. If there is no slack, assume we'll
     * continue to run early.
     */

Based on GTFS data I've seen, I believe this tries to address the behavior that happens when an agency arbitrarily splits a trip into two (e.g., to change headsigns) but there isn't a layover in the schedule, and therefore the bus doesn't hold and continues to run early.  If there is a planned break, the driver is more likely to hold and wait to depart until the scheduled departure time.

An alternate rule is that early arrivals *never* get propagated across trips, but it risks switching predictions from "early" to "ontime" or "no data" arbitrarily in the middle of a passenger-perceived trip when the agency arbitrarily splits a trip in their GTFS.

Reply:
LGTM but I have only opinion on this subject, and we should better listen to experience. Onboard maybe 4 or 5 agencies on this, to see what is their usual business rule with that. Maybe some consumers have some insight on this, if they do statistics on the real-time they are processing.

Reply:
Thanks, makes sense.  I'll ask for comments when I open a PR on the main repo.